File: src/main/java/no/nav/familie/integrasjoner/aktør/AktørController.java
package no.nav.familie.integrasjoner.aktør;

import no.nav.familie.integrasjoner.personopplysning.domene.AktørId;
import no.nav.familie.ks.kontrakter.sak.Ressurs;
import no.nav.security.token.support.core.api.ProtectedWithClaims;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.constraints.NotNull;
import java.util.Map;

@RestController
@ProtectedWithClaims(issuer = "azuread")
@RequestMapping("/api/aktoer")
public class AktørController {

    private AktørService aktørService;

    @Autowired
    AktørController(AktørService aktørService) {
        this.aktørService = aktørService;
    }

    @GetMapping("v1")
    public ResponseEntity<Ressurs> getAktørIdForPersonIdent(@NotNull @RequestHeader(name = "Nav-Personident") String personIdent) {
        return ResponseEntity
                .ok()
                .body(Ressurs.Companion.success(
                        Map.of("aktørId", aktørService.getAktørId(personIdent)), "Hent aktør for personident OK")
                );
    }

    @GetMapping(path = "v1/fraaktorid")
    public ResponseEntity<Ressurs> getPersonIdentForAktørId(@NotNull @RequestHeader(name = "Nav-Aktorid") String aktørId) {
        return ResponseEntity
                .ok()
                .body(Ressurs.Companion.success(
                        Map.of("personIdent", aktørService.getPersonIdent(new AktørId(aktørId))), "Hent personIdent for aktør OK")
                );
    }
}

File: src/main/java/no/nav/familie/integrasjoner/aktør/AktørController.kt
package no.nav.familie.integrasjoner.aktør

import no.nav.familie.integrasjoner.personopplysning.domene.AktørId
import no.nav.familie.kontrakter.felles.Ressurs
import no.nav.familie.kontrakter.felles.Ressurs.Companion.success
import no.nav.familie.kontrakter.felles.Tema
import no.nav.security.token.support.core.api.ProtectedWithClaims
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@ProtectedWithClaims(issuer = "azuread")
@RequestMapping("/api/aktoer")
class AktørController(
    private val aktørService: AktørService,
) {
    /**
     * Denne brukes av familie-ef-mottak for å finne behandlende enhet.
     * Kan fjernes når eller hvis det blir mulig å assosiere journalposter med kun personident.
     */
    @PostMapping("v2/fraaktorid/{tema}")
    fun finnPersonIdentForAktørId(
        @RequestBody(required = true) aktørId: String,
        @PathVariable tema: Tema,
    ): ResponseEntity<Ressurs<Map<String, String>>> =
        ResponseEntity.ok(
            success(
                mapOf("personIdent" to aktørService.getPersonIdentFraPdl(AktørId(aktørId), tema)),
                "Hent aktør for personident OK",
            ),
        )
}
